bioimageio/spec/utils/build_spec.py

```diff
@@ -0,0 +1,251 @@
+        assert offset is not None
+        shape_description = {
+            'reference_input': reference_input,
+            'scale': scale,
+            'offset': offset
+        }
+
+    kwargs = {}
+    if postprocessing is not None:
+        kwargs['postprocessing'] = postprocessing
+    if halo is not None:
+        kwargs['halo'] = halo
+
+    outputs = spec.raw_nodes.OutputTensor(
+        name='output' if name is None else name,
+        data_type=str(test_out.dtype),
+        axes='bczyx' if test_out.ndim == 5 else 'bcyx',
+        shape=shape_description,
+        **kwargs
+    )
+    return outputs
+
+
+# TODO can we pattern match to decide if we have a url or doi?
+def _build_cite(cite):
+    citation_list = [
+        spec.raw_nodes.CiteEntry(text=k, url=v) for k, v in cite.items()
+    ]
+    return citation_list
+
+
+# NOTE does not support multiple input / output tensors yet
+# to implement this we should wait for 0.4.0, see also
+# https://github.com/bioimage-io/spec-bioimage-io/issues/70#issuecomment-825737433
+def build_spec(
+    # model specific required
+    source: str,
+    model_kwargs: Dict[str, Union[int, float, str]],
+    weight_uri: str,
+    test_inputs: List[str],
+    test_outputs: List[str],
+    # general required
+    name: str,
+    description: str,
+    authors: List[str],
+    tags: List[str],
+    license: str,
+    documentation: str,
+    covers: List[str],
+    dependencies: str,
+    cite: Dict[str, str],
+    root: Optional[str] = None,
+    # model specific optional
+    weight_type: Optional[str] = None,
+    sample_inputs: Optional[str] = None,
+    sample_outputs: Optional[str] = None,
+    # tensor specific
+    input_name: Optional[str] = None,
+    input_step: Optional[List[int]] = None,
+    input_min_shape: Optional[List[int]] = None,
+    output_name: Optional[str] = None,
+    output_reference: Optional[str] = None,
+    output_scale: Optional[List[int]] = None,
+    output_offset: Optional[List[int]] = None,
+    halo: Optional[List[int]] = None,
+    preprocessing: Optional[List[Dict[str, Dict[str, Union[int, float, str]]]]] = None,
+    postprocessing: Optional[List[Dict[str, Dict[str, Union[int, float, str]]]]] = None,
+    # general optional
+    git_repo: Optional[str] = None,
+    attachments: Optional[Dict[str, Union[str, List[str]]]] = None,
+    packaged_by: Optional[List[str]] = None,
+    run_mode: Optional[str] = None,
+    parent: Optional[str] = None,
+    config: Optional[Dict[str, Any]] = None,
+):
+    """
+    """
+    #
+    # generate the model specific fields
+    #
+
+    # check the test inputs and auto-generate input/output description from test inputs/outputs
+    for test_in, test_out in zip(test_inputs, test_outputs):
+        test_in, test_out = _ensure_uri(test_in, root), _ensure_uri(test_out, root)
+        test_in, test_out = np.load(test_in), np.load(test_out)
+    inputs = _get_input_tensor(test_in, input_name, input_step, input_min_shape, preprocessing)
```

Comment:
when running the test I get `inputs.axes = "bcyx"` and `input.shape = (10000, 1)`... so something is off

Reply:
Yes, I know what's going on, I am fixing it now.

Reply:
feel free to merge in https://github.com/bioimage-io/python-bioimage-io/pull/66 at any time that suits you. I just realized that the test dir was a bit out of order...